packages/ts/src/components/nested-donut/index.ts

```diff
@@ -0,0 +1,197 @@
+import { Selection } from 'd3-selection'
+import { arc } from 'd3-shape'
+import { hierarchy, partition } from 'd3-hierarchy'
+import { scaleLinear, ScaleLinear } from 'd3-scale'
+import { group } from 'd3-array'
+
+// Core
+import { ComponentCore } from 'core/component'
+import { SeriesDataModel } from 'data-models/series'
+
+// Types
+import { VerticalAlign } from 'types/text'
+
+// Utils
+import { getColor, getHexValue } from 'utils/color'
+import { getString, getValue, isNumber, isNumberWithinRange } from 'utils/data'
+import { smartTransition } from 'utils/d3'
+import { wrapSVGText } from 'utils/text'
+
+// Local Types
+import { NestedDonutDirection, NestedDonutSegment, defaultLayerSettings, NestedDonutLayer } from './types'
+
+// Config
+import { NestedDonutConfig, NestedDonutConfigInterface } from './config'
+
+// Modules
+import { createArc, updateArc, removeArc } from './modules/arc'
+import { createLabel, updateLabel, removeLabel } from './modules/label'
+
+// Styles
+import * as s from './style'
+
+export class NestedDonut<Datum> extends ComponentCore<
+Datum[],
+NestedDonutConfig<Datum>,
+NestedDonutConfigInterface<Datum>
+> {
+  static selectors = s
+  static cssVariables = s.variables
+  config: NestedDonutConfig<Datum> = new NestedDonutConfig()
+  datamodel: SeriesDataModel<Datum> = new SeriesDataModel()
+
+  arcBackground: Selection<SVGGElement, unknown, SVGGElement, unknown>
+  arcGroup: Selection<SVGGElement, unknown, SVGGElement, unknown>
+  centralLabel: Selection<SVGTextElement, unknown, SVGGElement, unknown>
+  centralSubLabel: Selection<SVGTextElement, unknown, SVGGElement, unknown>
+
+  arcGen = arc<Partial<NestedDonutSegment<Datum>>>()
+  colorScale: ScaleLinear<string, string> = scaleLinear()
+
+  events = { }
+
+  constructor (config?: NestedDonutConfigInterface<Datum>) {
+    super()
+    if (config) this.config.init(config)
+    this.arcBackground = this.g.append('g')
+    this.arcGroup = this.g.append('g')
+    this.centralLabel = this.g.append('text')
+      .attr('class', s.centralLabel)
+    this.centralSubLabel = this.g.append('text')
+      .attr('class', s.centralSubLabel)
+  }
+
+  private getHierarchyData (layers: NestedDonutLayer[]): NestedDonutSegment<Datum>[] {
```

Comment:
I don't have a super strong option on it, but I prefer to prefix private methods with an underscore `_`. This makes it easier to read through the code because you can immediately see which method calls are purely internal.